iceoryx_posh/include/iceoryx_posh/internal/popo/wait_set.inl

```diff
@@ -317,10 +317,77 @@ inline void WaitSet<Capacity>::acquireNotifications(const WaitFunction& wait) no
     }
     else if (!notificationVector.empty())
     {
-        m_activeNotifications = algorithm::uniqueMergeSortedContainers(notificationVector, m_activeNotifications);
+        uniqueMergeSortedNotificationVector(notificationVector, m_activeNotifications)
+            .and_then([&](auto& mergedNotifications) { m_activeNotifications = mergedNotifications; })
+            .or_else([](auto& error) {
+                if (error == NotificationVectorMergeError::MERGED_VECTOR_OVERFLOW)
+                {
+                    IOX_LOG(WARN) << "Overflow when merging new notifications with already active notification! "
+                                     "Consider increasing the CMake switch 'IOX_INTERNAL_MAX_NUMBER_OF_NOTIFIERS'";
+                }
+            });
     }
 }
 
+template <uint64_t Capacity>
+inline cxx::expected<ConditionListener::NotificationVector_t, NotificationVectorMergeError>
```

Comment:
Please do not adjust the API here. This is an internal algorithm of the waitset which is guaranteed to work due to design.
Another thing is, they callee can ensure that this works correctly by providing two vectors where the sum of all elements does not exceed the capacity.

Furthermore, we can also state in the internal doc, merges two vectors until capacity is reached so that user knows that it should be ensured by other means that this does not truncate.

Reply:
@elfenpiff Sure, got it. Reason is stated [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1800#discussion_r1028196890). As this method is now private API of the `WaitSet` I don't think we'll need any doxygen or docs about it.